service/config.go

```diff
@@ -124,9 +145,14 @@ func (cfg *Config) validateService() error {
 		// Validate pipeline receiver name references.
 		for _, ref := range pipeline.Receivers {
 			// Check that the name referenced in the pipeline's receivers exists in the top-level receivers.
-			if cfg.Receivers[ref] == nil {
-				return fmt.Errorf("pipeline %q references receiver %q which does not exist", pipelineID, ref)
+			if cfg.Receivers[ref] != nil {
```

Comment:
Is there a performance gain if you use the optional second return value (bool) from the map lookup?

```suggestion
			if _, ok := cfg.Receivers[ref]; ok {
```

Reply:
Good point. I think your suggestion is more idiomatic either way.

Reply:
Fixed